darkdetect/__init__.py

```diff
@@ -4,41 +4,59 @@
 #  Distributed under the terms of the 3-clause BSD License.
 #-----------------------------------------------------------------------------
 
-__version__ = '0.7.1'
+
+__version__: str = '0.7.1'
+
 
 import sys
 import platform
+from typing import Callable, Type
+
+from .base import BaseListener
+Listener: Type[BaseListener]
+
+#
+# Import correct the listener for the given OS
+#
 
 def macos_supported_version():
-    sysver = platform.mac_ver()[0] #typically 10.14.2 or 12.3
-    major = int(sysver.split('.')[0])
+    sysver: str = platform.mac_ver()[0]  # typically 10.14.2 or 12.3
+    major: int = int(sysver.split('.')[0])
     if major < 10:
         return False
-    elif major >= 11:
+    if major >= 11:
         return True
-    else:
-        minor = int(sysver.split('.')[1])
-        if minor < 14:
-            return False
-        else:
-            return True
-
-if sys.platform == "darwin":
-    if macos_supported_version():
-        from ._mac_detect import *
-    else:
-        from ._dummy import *
-elif sys.platform == "win32" and platform.release().isdigit() and int(platform.release()) >= 10:
-    # Checks if running Windows 10 version 10.0.14393 (Anniversary Update) OR HIGHER. The getwindowsversion method returns a tuple.
-    # The third item is the build number that we can use to check if the user has a new enough version of Windows.
-    winver = int(platform.version().split('.')[2])
-    if winver >= 14393:
-        from ._windows_detect import *
-    else:
-        from ._dummy import *
+    return int(sysver.split('.')[1]) >= 14
+
+if sys.platform == "darwin" and macos_supported_version():
+    from ._mac_detect import *
+    Listener = MacListener
+# If running Windows 10 version 10.0.14393 (Anniversary Update) OR HIGHER.
+# The getwindowsversion method returns a tuple.
+# The third item is the build number that we can use to check if the user has a new enough version of Windows.
+elif sys.platform == "win32" and platform.release().isdigit() and int(platform.release()) >= 10 and \
+        int(platform.version().split('.')[2]) >= 14393:
+    from ._windows_detect import *
+    Listener = WindowsListener
 elif sys.platform == "linux":
     from ._linux_detect import *
+    Listener = GnomeListener
 else:
     from ._dummy import *
+    Listener = DummyListener
+
+#
+# Common shortcut functions
+#
+
+def isDark():
+    return theme() == "Dark"
+
+def isLight():
+    return theme() == "Light"
```

Comment:
This violates the existing API. In case detection is not supported, both `isDark()` and `isLight()` should return `None`.

Reply:
Interesting, I assumed that was a bug and that only `theme()` was meant to return None if not supported based on the README. I will revert this then.

Reply:
Actually, I have a question in this case:
https://github.com/albertosottile/darkdetect/blob/3347df3e95e8a96118b3ae7229a8fe699701159c/darkdetect/_windows_detect.py#L73
Windows (is the only OS that) checks if `theme()` is `None` before doing this `==`. Why? Given that it passed the gates in `__init__.py`, isn't it marked supported? I.e. is this an error and thus we are deeming it as unsupported and thus returning `None`? Or something else? In this case, why do `Linux` and `Mac` not have the same conditionals, or is it just no need has arisen to put one in? In which case, would all OSes be supported by something like:
```python
def isDark() -> Optional[str]"
  t = theme()
  if t is not None:
    return t == "Dark"
```
I can just ignore this and return the original logic- I just prefer consolidation of these lines unless there is a reason not to, since they seem like they *should* all be the same.

Reply:
> Windows (is the only OS that) checks if theme() is None before doing this ==. Why?

I think this is the reason:
```
    except FileNotFoundError:
        # some headless Windows instances (e.g. GitHub Actions or Docker images) do not have this key
        return None
```

The detection in `__init__` is only meant to check for OS support, but sometimes there are further situations in which it is not possible to determine the theme even though the OS supports this, in principle.

> In this case, why do Linux and Mac not have the same conditionals,

Because on macOS the `theme` function always works if the version gates are passed. I think the Linux code is developed with the same assumption, but I am not 100% sure that's actually the case because I am not a Linux user.

> In this case, why do Linux and Mac not have the same conditionals,

Fine for me, I also value consolidation. But I would still like to have `None` returned by `theme()`, `isLight()`, and `isDark()` when `darkdetect` is unable to correctly detect the theme.